### PR TITLE
feat: support SQL Read Replica and Private Google Access

### PR DESCRIPTION
**Pull request by poom**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [X] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [X] I have typed an adequate description that explains **why** I am making this change.
- [X] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

- Support Private IP - It contains 3 modules - `google_compute_global_address`, `google_compute_network`, and `google_service_networking_connection`
- Update google-sql-database-instance to support read replica

---
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-sql/3)

### CODE REVIEW

**Review by maetolay — APPROVED** [review]

File: examples/create_mysql_instance_with_private_ip/providers.tf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+provider "google" {
+  alias       = "vpc"
+  project     = var.google_project
+  region      = var.google_region
+  credentials = var.google_credentials
+}
+
+#provider "google-beta" {
```
Comment (MXfive) [inline examples/create_mysql_instance_with_private_ip/providers.tf]: Can we remove this?

File: examples/create_mysql_instance_with_public_ip/remote-backend.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+terraform {
```
Comment (MXfive) [inline examples/create_mysql_instance_with_public_ip/remote-backend.tf]: Can delete the whole file instead?

File: examples/mysql_instance_with_read_replica/providers.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+provider "google" {
+  alias       = "vpc"
+  project     = var.google_project
+  region      = var.google_region
+  credentials = var.google_credentials
+}
+
+#provider "google-beta" {
```
Comment (MXfive) [inline examples/mysql_instance_with_read_replica/providers.tf]: Can delete these instead?

File: examples/mysql_instance_with_read_replica/remote-backend.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+terraform {
```
Comment (MXfive) [inline examples/mysql_instance_with_read_replica/remote-backend.tf]: Can delete the whole file instead?

File: modules/google_compute_global_address/main.tf [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0.0"
+    }
+  }
+}
+
+resource "google_compute_global_address" "default_network" {
+  #  provider = google-beta
```
Comment (MXfive) [inline modules/google_compute_global_address/main.tf]: Delete this?

File: modules/google_compute_network/main.tf [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0.0"
+    }
+  }
+}
+
+resource "google_compute_network" "vpc_network" {
+  #  provider = google-beta
```
Comment (MXfive) [inline modules/google_compute_network/main.tf]: Delete this

File: modules/google_service_networking_connection/main.tf [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0.0"
+    }
+  }
+}
+
+resource "google_service_networking_connection" "network" {
+  #  provider = google-beta
```
Comment (MXfive) [inline modules/google_service_networking_connection/main.tf]: Delete this

**Review by MXfive — CHANGES_REQUESTED** [review]

**Review by MXfive — APPROVED** [review]

**Pull request merged**